app/views/hit_count/_include.html.erb

```diff
@@ -0,0 +1,12 @@
+<% unless @work.in_unrevealed_collection? ||
+          @work.hidden_by_admin? ||
+          !@work.posted? ||
+          current_user&.is_author_of?(@work) %>
+  <% content_for :footer_js do %>
+    <script>
+      $j(document).on("loadedCSRF", function() {
+        $j.post("<%= work_hit_count_path(@work, format: :json) %>")
```

Comment:
Yeah, if you have the right CSRF tokens you can make this POST request from anywhere. (You can even make it for works that don't exist.)

I can repeat most parts of the `unless` check in `RedisHitCounter#save_recent_counts`, but I can't check `current_user.is_author_of?(@work)`. (And as I mentioned, changes in visibility may result in valid hits being skipped/invalid hits being added.)

Reply:
If we repeat the `unless` check in `save_recent_counts`:

- No `current_user` check: you could easily game this before by logging out and viewing your own work (before we started logged-out full-page caching), so we can let this slide.
- Changes in visibility (i.e. works become unrevealed and we discard the last max 30m of hits): I'm fine with this accuracy trade-off, considering the alternative is to load the work and check in the hits count controller, which you've set up tests to avoid for performance reasons.

Let's see what other reviewers think.

Reply:
For unrevealed works, we really only stopped counting hits because users who posted directly to an unrevealed collection would freak out when they had 1 or 2 hits instead of 0 -- I can't remember ever having a complaint that was like, "My work had 2394 when I put it in an unrevealed collection, but now it has 2576." So I am fairly indifferent regarding what we do for works that change status.

Reply:
I added a check to `save_batch_hit_counts` (the method used by `save_recent_counts` to actually save the hit counts), since it sounds like that's the consensus.

It occurred to me that there might technically be another way to game it: a guest could add a hit to a restricted work if they know the ID. But I'm not sure how to resolve that without having the `HitCountController` hit the database for at least some requests.

Reply:
> a guest could add a hit to a restricted work if they know the ID

We decided not to check for this. Guests are still subjected to the one hit per IP per 24h restriction.

Reply:
I just remember that this is pretty similar to downloads: we check for unrevealed and hidden works, but we don't check for restricted works, also because of nginx caching.